lib/spack/spack/binary_distribution.py

```diff
@@ -0,0 +1,491 @@
+                  key=None):
+    """
+    Build a tarball from given spec and put it into the directory structure
+    used at the mirror (following <tarball_directory_name>).
+    """
+    # set up some paths
+    tarfile_name = tarball_name(spec, '.tar.gz')
+    tarfile_dir = join_path(outdir, "build_cache",
+                            tarball_directory_name(spec))
+    tarfile_path = join_path(tarfile_dir, tarfile_name)
+    mkdirp(tarfile_dir)
+    spackfile_path = os.path.join(
+        outdir, "build_cache", tarball_path_name(spec, '.spack'))
+    if os.path.exists(spackfile_path):
+        if force:
+            os.remove(spackfile_path)
+        else:
+            return NoOverwriteException, spackfile_path
+    # need to copy the spec file so the build cache can be downloaded
+    # without concretizing with the current spack packages
+    # and preferences
+    spec_file = join_path(spec.prefix, ".spack", "spec.yaml")
+    specfile_name = tarball_name(spec, '.spec.yaml')
+    print specfile_name
+    specfile_path = os.path.realpath(
+        join_path(outdir, "build_cache", specfile_name))
+    print specfile_path
+    indexfile_path = join_path(outdir, "build_cache", "index.html")
+    if os.path.exists(specfile_path):
+        if force:
+            os.remove(specfile_path)
+        else:
+            return NoOverwriteException(specfile_path)
+    # make a copy of the install directory to work with
+    prefix = join_path(outdir, os.path.basename(spec.prefix))
+    if os.path.exists(prefix):
+        shutil.rmtree(prefix)
+    install_tree(spec.prefix, prefix)
+
+    # create info for later relocation and create tar
+    write_buildinfo_file(prefix)
+
+    # optinally make the paths in the binaries relative to each other
+    # in the spack install tree before creating tarball
+    if rel:
+        make_package_relative(prefix)
+    # create compressed tarball of the install prefix
+    with closing(tarfile.open(tarfile_path, 'w:gz')) as tar:
+        tar.add(name='%s' % prefix,
+                arcname='%s' % os.path.basename(prefix))
+    # remove copy of install directory
+    shutil.rmtree(prefix)
+
+    # get the sha256 checksum of the tarball
+    checksum = checksum_tarball(tarfile_path)
+
+    # add sha256 checksum to spec.yaml
+    spec_dict = {}
+    with open(spec_file, 'r') as inputfile:
+        content = inputfile.read()
+        spec_dict = yaml.load(content)
+    bchecksum = {}
+    bchecksum['algo'] = 'sha256'
+    bchecksum['hash'] = checksum
+    spec_dict['binary_cache_checksum'] = bchecksum
+    with open(specfile_path, 'w') as outfile:
+        outfile.write(yaml.dump(spec_dict))
+    signed = False
+    if not yes_to_all:
+        # sign the tarball and spec file with gpg
+        try:
+            sign_tarball(yes_to_all, key, force, specfile_path)
+            signed = True
+        except (NoGpgException, PickKeyException, NoKeyException) as e:
```

Comment:
In this case you don't have to surround with try/except, the exception will terminate and be caught in the caller (generally speaking when you have a try/except clause and the except just has a ```raise``` then you don't need to wrap with a try/except